lib/matplotlib/axes/_axes.py

```diff
@@ -3616,7 +3690,8 @@ def scatter(self, x, y, s=20, c=None, marker='o', cmap=None, norm=None,
             if facecolors is not None:
                 c = facecolors
             else:
-                c = 'b'  # the original default
+                color_cycle = self._get_lines.color_cycle
```

Comment:
:-1: on scatter using color cycle.  This has come up before and the consensus was that scatter should be independent of color cycle because it is as scalar mappable.  One of two  reasons to use `scatter` over a `plot` with no line is if you want to map the color of the markers to an array (the other is to scale the size of the markers (and maybe the rotation?)).

This is also an major API break, if we _do_ want to make `scatter` sometimes hook into the color cycle it should be done on the `color_overhaul` branch for the 2.0 release.